### PR TITLE
Cricket mapping patch

### PR DESCRIPTION
**Pull request by AsciiSquid**
## About The Pull Request
Some tweaks to the Solgov-Cricket to make it more up-to-date with modern shiptesting standards.
Full name has been changed from Solgov-Issued Cricket Scout to Cricket-Class Solgov Intel Scout so the class name comes first on the ship list, making it easier to read for.
- Airlocks relabelled to their respective departments
- Dormitory clothing vendors replaced with solgov fatigues locker
- Bridge includes external window shutter controls
- Cargo bay reorganised to fit express supply units
- Starter mining equipment includes GPS
- Added fridge of ingredients in the storeroom
- Renault the fox added to the Solgov office
- Additional chairs added throughout the ship to reduce workplace hazards as a result of shuttle turbulence
-  Aft-Hall airlock made smaller to improve cycle speeds
- Telecomms engineering hardsuit replaced with Solgov vacuum suit
- Chemistry bag added to the infirmary pantry
newcricket
Additionally, added a mapping helper object `/obj/machinery/suit_storage_unit/inherit`. A suit storage unit which will pick up any spacesuit equipment left in it's turf on mapload similar to closets.

## Why It's Good For The Game
Makes map gooder

## Changelog
:cl:
add: added suit storage mapping helper
tweak: tweaked design of the cricket-class
/:cl:

### CODE REVIEW

File: code/game/machinery/suit_storage_unit.dm [hunk 1]
```diff
@@ -517,3 +524,27 @@
 		I.play_tool_sound(src, 50)
 		visible_message("<span class='notice'>[usr] pries open \the [src].</span>", "<span class='notice'>You pry open \the [src].</span>")
 		open_machine()
+
+// Mapping helper unit takes whatever lies on top of it
+/obj/machinery/suit_storage_unit/inherit/Initialize(mapload)
+	. = ..()
+	wires = new /datum/wires/suit_storage_unit(src)
```
Comment (MarkSuckerberg) [inline code/game/machinery/suit_storage_unit.dm]: ?

File: code/game/machinery/suit_storage_unit.dm [hunk 2]
```diff
@@ -517,3 +524,27 @@
...
+	if(mapload)
+		addtimer(CALLBACK(src, .proc/inherit_contents), 0)
```
Comment (MarkSuckerberg) [inline code/game/machinery/suit_storage_unit.dm]: Should probably move this to LateInitialize, make sure that the original initialize here returns the lateinit hint

File: code/game/machinery/suit_storage_unit.dm [hunk 3]
```diff
@@ -517,3 +524,27 @@
...
+
+/obj/machinery/suit_storage_unit/inherit/proc/inherit_contents()
+	var/turf/T = src.loc
+	for(var/atom/movable/AM in T)
+		if(istype(AM, /obj/item/clothing/suit/space))
```
Comment (MarkSuckerberg) [inline code/game/machinery/suit_storage_unit.dm]:
```suggestion
		if(istype(AM, /obj/item/clothing/suit/space) && !suit)
```

File: code/game/machinery/suit_storage_unit.dm [hunk 4]
```diff
@@ -517,3 +524,27 @@
...
+			AM.forceMove(src)
+			suit = AM
+		else if(istype(AM, /obj/item/clothing/head/helmet/space))
```
Comment (MarkSuckerberg) [inline code/game/machinery/suit_storage_unit.dm]:
```suggestion
		else if(istype(AM, /obj/item/clothing/head/helmet/space) && !helmet)
```

File: code/game/machinery/suit_storage_unit.dm [hunk 5]
```diff
@@ -517,3 +524,27 @@
...
+			AM.forceMove(src)
+			helmet = AM
+		else if(istype(AM, /obj/item/clothing/mask))
```
Comment (MarkSuckerberg) [inline code/game/machinery/suit_storage_unit.dm]:
```suggestion
		else if(istype(AM, /obj/item/clothing/mask) && !mask)
```

File: code/game/machinery/suit_storage_unit.dm [hunk 6]
```diff
@@ -517,3 +524,27 @@
...
+			AM.forceMove(src)
+			mask = AM
+		else if(istype(AM, /obj/item))
```
Comment (MarkSuckerberg) [inline code/game/machinery/suit_storage_unit.dm]:
```suggestion
		else if(istype(AM, /obj/item) && !storage)
```

**Review by MarkSuckerberg — CHANGES_REQUESTED** [review]

File: code/game/machinery/suit_storage_unit.dm (hunk 1 above)
Comment (AsciiSquid) [reply code/game/machinery/suit_storage_unit.dm]: Copy and paste responsibly kids

File: code/game/machinery/suit_storage_unit.dm (hunk 2 above)
Comment (AsciiSquid) [reply code/game/machinery/suit_storage_unit.dm]: Borrowed the method from `/obj/structure/closet/Initialize(mapload)` but I can certainly use that instead.

**Review by AsciiSquid — COMMENTED** [review]
Will apply fixes in upcoming commit

**Pull request merged**